src/worker/validator.py

```diff
@@ -29,7 +30,9 @@ def _run_and_optimize_model(self, data):
 
         elif self.optimize_strategy == 'GAN':
             total_loss = 0
-            for network_name in self.model.network_names:
+            for optimizer_name in self.optimizers.keys():
+                self.optimizers[optimizer_name].zero_grad()
```

Comment:
These lines are fixed in another branch (will be a PR soon). Also, `Validator` does not hold the attribute `optimizers`.
(I'll remove it on rebasing)